packages/web/src/features/translation/TranslationView.tsx

```diff
@@ -312,9 +313,59 @@ export default function TranslationView() {
     isFlagEnabled('comments') &&
     !!userCan('read', { type: 'Language', id: language });
 
+  const flash = useFlash();
+
+  const glossesAsDisplayed = useMemo(
+    () =>
+      targetGlossesQuery.data?.data.map((targetGloss) => ({
+        wordId: targetGloss.wordId,
+        glossAsDisplayed:
+          targetGloss.gloss ||
+          targetGloss.suggestions[0] ||
+          targetGloss.machineGloss,
+        state: targetGloss.state,
+      })),
+    [targetGlossesQuery.data]
+  );
+
+  const approveAllGlossesMutation = useMutation({
+    mutationFn: async () => {
+      if (glossesAsDisplayed) {
+        const data = Object.fromEntries(
+          glossesAsDisplayed
+            .filter(
+              ({ glossAsDisplayed, state }) =>
+                glossAsDisplayed !== undefined &&
+                state === GlossState.Unapproved
+            )
+            .map(({ wordId, glossAsDisplayed }) => [
+              wordId,
+              { gloss: glossAsDisplayed, state: GlossState.Approved },
+            ])
+        );
+        await apiClient.verses.updateVerseGlosses(verseId, language, {
+          data,
+        });
+      }
+    },
+    onSuccess: async () => {
+      await targetGlossesQuery.refetch();
```

Comment:
The better way to reload state is use `queryClient.invalidateQueries`. That's because there could be multiple places that use the same query key to load the same data from the cache and `invalidateQueries` handles all of them.

Reply:
Done.